src/plugins/data/public/antlr/dql/code_completion.ts

```diff
@@ -45,38 +48,26 @@ const findFieldSuggestions = (indexPattern: IndexPattern) => {
       return idxField.name;
     });
 
-  const fieldSuggestions: Array<{
-    text: string;
-    type: monaco.languages.CompletionItemKind;
-  }> = fieldNames.map((field: string) => {
+  const fieldSuggestions: QuerySuggestion[] = fieldNames.map((field: string) => {
     return {
       text: field,
       type: monaco.languages.CompletionItemKind.Field,
       insertText: `${field}: `,
+      start: -1,
+      end: -1,
     };
   });
 
   return fieldSuggestions;
 };
 
-const getFieldSuggestedValues = async (
-  indexTitle: string,
-  fieldName: string,
-  currentValue: string,
-  http?: HttpSetup
-) => {
-  if (!http) return [];
-  return await http.fetch(`/api/opensearch-dashboards/suggestions/values/${indexTitle}`, {
-    method: 'POST',
-    body: JSON.stringify({ query: currentValue, field: fieldName, boolFilter: [] }),
-  });
-};
-
 const findValueSuggestions = async (
   index: IndexPattern,
```

Comment:
is this the indexpattern service or just object. because actually can just take the query object which now has the dataset object which has all the information you will need.

Reply:
This is the actual object, I believe its set by the query_editor.tsx function which comes from the dataset object.